### PR TITLE
Refactor creation of reservation series

### PR DESCRIPTION
**Pull request by joonatank**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."
 
- refactor: use a new series mutation to create recurring reservations
- fix: persistency of the page after creating a new recurring reservation (refresh / copy url).
- fix: show the number of reservations in a series without the ones that are removed.

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Manual testing: creating a new recurring reservation should work as before (except it's faster and the two linked bugs are fixed). 

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- [TILA-3288](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3288)
- [TILA-3352](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3352)
- [TILA-3392](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3392)

[TILA-3288]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3288?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TILA-3352]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3352?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TILA-3392]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/admin-ui/src/spa/my-units/recurring/RecurringReservationForm.tsx [hunk 1]
```diff
@@ -203,78 +219,90 @@ export function RecurringReservationForm({ reservationUnits }: Props) {
...
+    const skipDates = removedReservations
+      .concat(checkedReservations.reservations.filter((x) => x.isOverlapping))
+      .map((x) => x.date);
+
+    /* TODO do we need this check? probably should have it anyway (or have the mutation throw if there are no reservations)
     if (reservationsToMake.length === 0) {
       errorToast({ text: t(translateError("noReservations")) });
       return;
     }
+    */
```
Comment (joonatank) [inline apps/admin-ui/src/spa/my-units/recurring/RecurringReservationForm.tsx:250]: check? `checkedReservations.reservations`

File: apps/admin-ui/src/spa/my-units/recurring/hooks/useFilteredReservationList.tsx [hunk 2]
```diff
@@ -172,12 +149,35 @@ export function useFilteredReservationList({
...
+    const timeToMinutes = (time: string) => {
+      const [hours, minutes] = time
+        .split(":")
+        .map(Number)
+        .filter(Number.isFinite);
+      if (hours != null && minutes != null) {
+        return hours * 60 + minutes;
+      }
+      return 0;
+    };
```
Comment (joonatank) [inline apps/admin-ui/src/spa/my-units/recurring/hooks/useFilteredReservationList.tsx:165]: can move to helper?

**Review by vincit-matu — APPROVED** [review]
Fix commented stuff, but otherwise good to go 😎 👍

File: apps/admin-ui/src/spa/my-units/recurring/RecurringReservationDone.tsx [hunk 3]
```diff
@@ -35,88 +36,55 @@ const ReservationMadeSchema = z.object({
...
 
-  const reservationId = successes.map((x) => x.reservationPk).find(() => true);
+  const { recurringReservation } = data ?? {};
+  const reservations = filterNonNullable(recurringReservation?.reservations);
 
-  // TODO holidays not implemented
-  const holidays = 0;
+  // TODO use urlbuilder
+  const reservationPk = reservations[0]?.pk;
+  const reservationUrl = reservationPk ? `/reservations/${reservationPk}` : "";
```
[the remark this answers is not in the document]
Comment (joonatank) [reply apps/admin-ui/src/spa/my-units/recurring/RecurringReservationDone.tsx:69]: changed in same PR where rest of url builders are

**Pull request merged**